source/TaskQueue.h

```diff
@@ -0,0 +1,42 @@
+/* TaskQueue.h
+Copyright (c) 2022 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef TASK_QUEUE_H_
+#define TASK_QUEUE_H_
+
+#include <functional>
+#include <future>
+
+
+
+// Class for queuing up tasks that are to be executed in parallel by using
+// every thread available on the executing CPU.
+class TaskQueue {
+public:
+	// Initialize the threads used to execute the tasks.
+	static void Init();
```

Comment:
Making this a static class doesn't feel right; the class does seem to have quite a number of variables to manage.